### DISCUSSION TITLE
Reset Prometheus Metrics

### DISCUSSION THREAD

**Discussion by jheiselman** [vmware-tanzu/velero#6520] [Community support Q&A]
Have velero working just fine in our VMware Tanzu Kubernetes cluster. Enabled metrics and have them getting pulled into Prometheus and alerts working.

However, in our case, the number of partially failed backups does not seem to update properly. We had one back up job improperly configured and it failed. Once the configuration was corrected, we deleted the partially failed backup objects using `velero backup delete` but the counter in prometheus did not go back down and now we are stuck with a persistent alert that is not correct.

**Comment by mateusoliveira43** [comment]
I think this is a Prometheus design, counter only increases https://prometheus.io/docs/concepts/metric_types/#counter

Maybe changing the alert rule? (or change metric to be gauge so it can increase or decrease)

**Reply by jheiselman** [reply]
That may be, but the implementation choice of using a counter was made by the velero project. I have restarted the process to clear cause the counter to reset, but it still seems counterintuitive that the counters don't reflect the reality of what is seen by investigating with the `velero` command.

**Reply by mateusoliveira43** [reply]
So the `velero` command gives a different view of Prometheus ("failed backups since Prometheus is running"). Do not know how is the Velero code for this part, but if you have some freedom to modify things maybe try different views of Prometheus (last 3 hours, for example, or graphs, since only one failure occured, it would be a flat line, meaning the number of failures is not growing) and different alert rules
